File: data/src/main/java/home/oleg/placesnearme/data/model/PhotoDbEntity.java
package home.oleg.placesnearme.data.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;

import home.oleg.placenearme.models.Photo;

import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity(foreignKeys = @ForeignKey(entity = DetailedVenueDbEntity.class,
        parentColumns = "id",
        childColumns = "venueId",
        onDelete = CASCADE))
public class PhotoDbEntity {

    @PrimaryKey(autoGenerate = true)
    private Long id;

    @ColumnInfo(index = true)
    private String venueId;

    @Embedded
    private Photo photo;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getVenueId() {
        return venueId;
    }

    public void setVenueId(String venueId) {
        this.venueId = venueId;
    }

    public Photo getPhoto() {
        return photo;
    }

    public void setPhoto(Photo photo) {
        this.photo = photo;
    }
}

File: core-data/src/main/java/home/oleg/placesnearme/data/model/PhotoDbEntity.kt
package home.oleg.placesnearme.data.model


import home.oleg.placenearme.models.Photo

import androidx.room.ColumnInfo
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.CASCADE
import androidx.room.PrimaryKey

@Entity(foreignKeys = [ForeignKey(entity = DetailedVenueDbEntity::class,
        parentColumns = arrayOf("id"),
        childColumns = arrayOf("venueId"),
        onDelete = CASCADE)])
data class PhotoDbEntity(
        @PrimaryKey(autoGenerate = true)
        var id: Long = 0,
        @ColumnInfo(index = true)
        var venueId: String = "",
        @Embedded
        var photo: Photo)

